### PR TITLE
[Feature] Low memory warning

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Catch "low memory warning" and provide the API to check if it has happened in last session: `Crashes.HasReceivedMemoryWarningInLastSession()`.

## Related PRs or issues

microsoft/appcenter#199
[AB#60785](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/60785)

### CODE REVIEW

File: Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs [hunk 1]
```diff
@@ -69,6 +70,46 @@ public void InstanceIsNotNull()
...
+            AppCenter.Instance.ApplicationSettings.SetValue(Crashes.PrefKeyMemoryWarning, true);
+            Crashes.Instance = new Crashes();
+            Crashes.Instance._memoryWarningHelper = mockMemoryWarningHelper.Object;
+
+            // Enable crash module.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            Crashes.Instance.ProcessPendingErrorsTask.Wait();
+
+            // Verify the memory warning is get and clean after the start.
```
Comment (annakocheshkova) [inline Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs]:
```suggestion
            // Verify the memory warning is received and cleaned after the start.
```

File: Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs [hunk 2]
```diff
@@ -69,6 +70,46 @@ public void InstanceIsNotNull()
...
+
+            // Enable crash module.
+            Crashes.SetEnabledAsync(true).Wait();
+            Crashes.Instance.OnChannelGroupReady(_mockChannelGroup.Object, string.Empty);
+            Crashes.Instance.ProcessPendingErrorsTask.Wait();
+
+            // Set memory warning.
+            AppCenter.Instance.ApplicationSettings.SetValue(Crashes.PrefKeyMemoryWarning, true);
+
+            // Verify clean the memory warning after disabling the crash module.
```
Comment (annakocheshkova) [inline Tests/Microsoft.AppCenter.Crashes.Test.Windows/CrashesTest.cs]:
```suggestion
            // Verify the memory warning is cleaned after disabling the crash module.
```

File: SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs [hunk 3]
```diff
@@ -443,6 +477,12 @@ private void ChannelSentLog(object sender, SentLogEventArgs e)
             }
         }
 
+        private void OnMemoryWarning(object sender, EventArgs e)
+        {
+            ApplicationSettings.SetValue(PrefKeyMemoryWarning, true);
+            AppCenterLog.Debug(LogTag, "The application received a low memory warning.");
```
Comment (annakocheshkova) [inline SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs]:
```suggestion
            AppCenterLog.Debug(LogTag, "The application received a low memory warning in the last session.");
```

**Comment by jaelim-ms** [comment]
Blocking this PR since #1069 was merged.
https://github.com/microsoft/appcenter-sdk-dotnet/pull/1069#discussion_r312572275
I tried to get a warning in Puppet with the current implementation (#1069) but I couldn't. Let me test this branch again.

**Comment by jaelim-ms** [comment]
It doesn't work in this branch as well.

**Comment by jaelim-ms** [comment]
Thanks @MatkovIvan for the link.
The issue was reported [here](https://github.com/MicrosoftDocs/winrt-api/issues/1196). Let me test this on a device.

**Pull request merged**